model/host/host.go

```diff
@@ -2783,6 +2783,19 @@ func FindHostWithHomeVolume(homeVolumeID string) (*Host, error) {
 	return FindOne(q)
 }
 
+// FindLatestTerminatedHostWithHomeVolume finds the most recently terminated host
+// associated with the specified home volume ID.
+func FindLatestTerminatedHostWithHomeVolume(homeVolumeID string) (*Host, error) {
+	q := db.Query(
+		bson.M{
+			StatusKey:       evergreen.HostTerminated,
```

Comment:
We have to also include the `StartedBy` key here to ensure we use the `{"started_by": 1, "status": 1}` index since just the status is not selective enough.